pkg/operator/ceph/cluster/osd/volumes.go

```diff
@@ -87,6 +99,56 @@ func getDeviceMapperVolume() (v1.Volume, v1.VolumeMount) {
 	return volume, volumeMounts
 }
 
+func (c *Cluster) getOSDKeyring() (v1.Volume, v1.VolumeMount) {
```

Comment:
This is not needed anymore, the secret `rook-ceph-admin-keyring` already exists, just attach it to init container.

Reply:
but it still need to run command `ceph -n client.admin auth get-or-create osd.$OSD_ID mon 'allow *' mgr 'allow *' osd 'allow *' -k  /etc/ceph/admin-keyring-store/keyring` ?

Reply:
Yes.

Reply:
I see the secret is mounted in the mount section of osd-0 pod describe 
```
rook-ceph-admin-keyring:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  rook-ceph-admin-keyring
    Optional:    false
```

but no file keyring is found when I checked inside osd-o pod
```
sh-4.4# pwd
/etc/ceph/admin-keyring-store
sh-4.4# ls
sh-4.4#
```
what I'm missing here?

Reply:
okay, I was looking to the wrong place, now I can see the secret.

But, I think still think for minikube we'll not able to get osd running after `minikube stop` `minikube start` as on reboot the symlink goes away according to [this](https://github.com/rook/rook/issues/4238#issuecomment-817728509 ) comment.